📝 Draft guide to adopting an ethical license

zspencer opened the pull request:
This is a first draft and is not ready to be linked from the home page yet.

It may provide a useful foil for fruitful discussion; and I'm particularly interested in feedback around the tone, truthfullness, and references I can cite for the different assertions being made.

See: https://github.com/EthicalSource/ethicalsource.dev/issues/22

zspencer committed:
📝 Draft guide to adopting an ethical license

This is a first draft and is not ready to be linked from the home page yet.

It may provide a useful foil for fruitful discussion.

See: See: https://github.com/EthicalSource/ethicalsource.dev/issues/22

nateberkopec commented on content/adopting-an-ethical-license.md:
```diff
@@ -0,0 +1,41 @@
+
+If you are starting a new project, you _may_ [review existing ethical licenses](./licenses/) and choose the one that most directly matches your goals.
+
+For maintainers considering adopting an ethical license on an existing project, be aware that this requires significant work. Each license is different, which makes a general guide difficult to write. Bear in mind that this is not legal advice, and we are not lawyers.
```
_may_ require. I personally don't think the MIT sublicense strategy is all that onerous, you just have to dot your i's and cross your t's. But GPL-licensed projects would yes obviously require a lot of work.

zspencer replied:
Even with MIT, it is a significant effort to relicense. I think it's much easier for new contributions to fall under an ethical license than it is to relicense past contributions, and agree with your point that clearly disambiguating the two pathways is a useful structural change.

nateberkopec commented on content/adopting-an-ethical-license.md:
```diff
@@ -0,0 +1,41 @@
+
+For maintainers considering adopting an ethical license on an existing project, be aware that this requires significant work. Each license is different, which makes a general guide difficult to write. Bear in mind that this is not legal advice, and we are not lawyers.
+
+Because there is significant social risk associated with relicensing, we recommend starting by centering an intent to preserve and further develop the trust between maintainers, contributors, and users of the project.
```
"centering an intent" is an unfamiliar turn of phrase for me, I think we could use plainer language here. But I love the point made.

zspencer replied:
Probably can replace it with "Prioritizing" or something.

nateberkopec commented on content/adopting-an-ethical-license.md:
```diff
@@ -0,0 +1,41 @@
+
+Because there is significant social risk associated with relicensing, we recommend starting by centering an intent to preserve and further develop the trust between maintainers, contributors, and users of the project.
+
+From there, an understanding of the authorship rights and license constraints will serve maintainers well. To do so, we recommend reaching a high degree of confidence in the precision and accuracy of your answers to the following questions:
```
I'd restate this whole paragraph as:

Before changing your project's licensing, you need to understand exactly how it works today. Consider:

zspencer replied:
I would prefer a strong assertion that people get clear answers to these two questions, as having good answers is the difference between a well-executed relicensing (or sub/dual-licensing) effort and one that draws valid complaints of erasure and appropriation.

nateberkopec replied:
IMO the social risks were clearly stated in the previous paragraph. 🤷‍♂

zspencer replied:
True, I'll tone it down in next draft :)

nateberkopec commented on content/adopting-an-ethical-license.md:
```diff
@@ -0,0 +1,41 @@
+
+Out of respect for these past authors, and to ensure adoption of ethical licenses proceed in a way that maintains the trust and integrity of the community; we believe maintainers _should_ acquire positive consent from all authors whose code remains within the project. Further, we recommend projects continue to conform to the constraints of their existing license until clear, positive, informed af ...
+
+## Existing License Constraints
```
I think it might be clearer if we lay out the two legal options available: adding an additional license _on top_ or _changing the license completely_.

zspencer replied:
Yeah I think describing the different mechanisms (relicensing, sub-licensing, dual-licensing) is probably worth calling out explicitely.

richardfontana commented on content/adopting-an-ethical-license.md:
```diff
@@ -0,0 +1,41 @@
+
+## Authorship Rights
+
+Authorship rights are often held by the original author of the contributed source code. Exceptions exist, such as when contributors explicitly assign authorship to the person(s) or organization(s) that maintain the project through a [Contributor License Agreement](https://en.wikipedia.org/wiki/Contributor_License_Agreement).
```
This is a little inaccurate. Possible rewrite:

## Authorship Rights

By default, the original author of the contributed source code owns copyright on the expressive content in that code. Copyright is the right to exclude others from certain activities relating to that expressive content, and is the main legal basis for software licensing. (Other kinds of intellectual property rights, such as patents, are also important in software licensing, but for simplicity we'll ignore them for purposes of this guide.) A license is a grant of permission to others to exercise rights that the owner could otherwise restrict. Typically, an employer owns copyright on software created by an employee. 

In a collaborative community software project, it is generally understood that each contributor is licensing their copyright under the license selected by the project. Exceptions exist, such as when a project requires contributors to sign a [Contributor License Agreement](https://en.wikipedia.org/wiki/Contributor_License_Agreement) requiring them to license copyright under more generous terms than the project license itself would require.

zspencer replied:
Thank you! This is a helpful reframing!

Aspie96 replied:
> In a collaborative community software project, it is generally understood that each contributor is licensing their copyright under the license selected by the project. Exceptions exist, such as when a project requires contributors to sign a [Contributor License Agreement](https://en.wikipedia.org/wiki/Contributor_License_Agreement) requiring them to license copyright under more generous terms than the project license itself would require.

@richardfontana, I think that most of the times, even when a CLA is present, contributions are still licensed under the same license as the project (to anybody receiving them) and, *in addition* they are licensed to a specific party according to the CLA (the contribution is effectively dual-licensed to them). Obviously that depends on the CLA but I think this is the most common scenario.

In practice it may not matter a lot, but I think it might be slightly more accurate?

Fryguy commented on content/adopting-an-ethical-license.md:
```diff
@@ -0,0 +1,41 @@
+
+The MIT License is one of the most widely used open source licenses, and also one of the most permissive. This makes it one of the easiest to relicense, but it still may require a significant amount of work.
+
+The core constraint of the MIT license is that the body of the license must be included in any copy or derivative works. This constraint persists so long as any code that was contributed under the license remains in the project _or_ every author with code in the project agrees to relicense their contributions under the new license.
```
I think this needs a slight wording change, since it sounds like you can't move forward without a 100% response.  For example, it sounds like code from author "A" can't be relicensed until code by author "B" is acknowledged.  However, if everyone except Author "B" accepts, you can still move forward, provided the code from Author "B" is removed.  (Maybe that's just my interpretation of this wording though)

zspencer replied:
My understanding is that you cannot _relicense_ without 100% response (or removal); so people who wish to adopt a non-MIT license without getting unanimous consent/removing non-consenting authors contributions must "sublicense;" which basically indicates that the license for _some_ of the code is MIT and _some_ of the code is a different license. See https://github.com/vcr/vcr/pull/805 for an example of sublicensing.

That said, I agree that that is the least clear part of the guide at this time :).

zspencer committed:
📝 Improve adoption guide per feedback

The main changes I made were:

1. Make `sublicensing` and `relicensing` the "signposts" for decision making
2. Switch from "you" to specify "maintainer" or "contributor"
3. Remove "Author"in favor or "contributor"
4. Cut. Words.

zspencer committed:
📝A bit more cleanup of the writing

zspencer committed:
Migrate adoption guide to blog

zspencer commented:
@CoralineAda  - I've moved this to the blog per your request. Let me know if there's anything else you'd like changed.

CoralineAda commented:
@zspencer Have you addressed all the comments, ready to merge? If so, please resolve the merge conflicts and we'll get this published. Thanks for all the hard work!

zspencer commented:
@CoralineAda - I do not resolve other peoples comments because only they can assert whether they have been resolved or not. The outstanding comments are marked as outdated; so I have addressed them as best as I am able to, but have not heard back from the original commentors.

This is the third or fourth time I've updated and moved things around to support shifting targets and structures. The changes to the structures are entirely reasonable, but it makes me feel like my efforts are unwanted and wasteful.


If the ethical source community feel this piece is worth publishing, I'm more than happy for anyone to resolve the conflicts and merge it. I'm not going to take the time to do it because the last few times have resulted in more requests for changes, and I only have limited capacity.

CoralineAda commented:
I think it’s a very valuable writeup. I’ll give it a final read tonight and then merge it unless something jumps out at me.


> On Jun 9, 2020, at 11:09 AM, Zee <notifications@github.com> wrote:
> 
> 
> @CoralineAda <https://github.com/CoralineAda> - I do not resolve other peoples comments because only they can assert whether they have been resolved or not. The outstanding comments are marked as outdated; so I have addressed them as best as I am able to, but have not heard back from the original commentors.
> 
> This is the third or fourth time I've updated and moved things around to support shifting targets and structures. The changes to the structures are entirely reasonable, but it makes me feel like my efforts are unwanted and wasteful.
> 
> If the ethical source community feel this piece is worth publishing, I'm more than happy for anyone to resolve the conflicts and merge it. I'm not going to take the time to do it because the last few times have resulted in more requests for changes, and I only have limited capacity.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/EthicalSource/ethicalsource.dev/pull/49#issuecomment-641405567>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAAFODHQFBZM53265IZ3MOLRVZNFJANCNFSM4LJ5JYYQ>.
>

zspencer commented:
Thank you!!! I really appreciate it :).
